app/controllers/orders_controller.rb

```diff
@@ -26,10 +29,13 @@ def create
     end
   end
 
-
   private
 
   def order_params
-    params.permit(:name, :address, :city, :state, :zip)
+    params.permit(:name, :address, :city, :state, :zip, :user_id)
+  end
+
+  def require_registered_user
+    render file: "shared/require_login" unless current_user
   end
```

Comment:
visitors have no link to check out, but this acts as a safeguard against manually entering the url.

Reply:
Good idea thinking of that, nice insurance policy.